api/src/context.rs

```diff
@@ -300,7 +312,32 @@ impl Context {
         let kvs = resource_iter
             .by_ref()
             .take(limit as usize)
-            .collect::<Result<_>>()?;
+            .collect::<Result<Vec<(StructTag, Vec<u8>)>>>()?;
+
+        // We should be able to do an unwrap here, otherwise the above db read would fail.
+        let resolver = self.state_view_at_version(version)?.into_move_resolver();
+
+        let kvs = kvs
```

Comment:
Some comments here about unwrapping resource groups into individual resources would be helpful 🙏